tests/test_assembly.py

```diff
@@ -808,20 +848,41 @@ def test_assemble_snp_bubble_stopbf(self, snp_bubble_structure):
 
         # do the labeling, but block the mutant with stop_bf
         stop_bf.count(mutant[HDN_L.pos + 1:HDN_L.pos + K + 1])
-        paths = lh.assemble_labeled_path(wildtype[:K], stop_bf)
-        assert len(paths) == 1
+        paths = asm.assemble(wildtype[:K], stop_bf)
 
+        assert len(paths) == 1
         assert any(utils._equals_rc(path, wildtype) for path in paths)
 
     # @pytest.mark.skip(reason='destroys your computer and then the world')
     def test_assemble_tandem_repeats(self, tandem_repeat_structure):
         # assemble one copy of a tandem repeat
         graph, repeat, tandem_repeats = tandem_repeat_structure
-
         lh = khmer._GraphLabels(graph)
+        asm = khmer.SimpleLabeledAssembler(lh)
+        paths = asm.assemble(repeat[:K])
 
-        paths = lh.assemble_labeled_path(repeat[:K])
         assert len(paths) == 1
         # There are K-1 k-mers spanning the junction between
         # the beginning and end of the repeat
         assert len(paths[0]) == len(repeat) + K - 1
+
+
+class TestJunctionCountAssembler:
+
+    def test_beginning_to_end_across_tip(self, right_tip_structure):
+        # assemble entire contig, ignoring branch point b/c of labels
```

Comment:
I understand what this test is testing for, but I'm still fuzzy on the junction counting and label implementation. Perhaps a brief explanation of how the assembler handles this simple example will help me wrap my mind around the more abstract concepts.